src/ngraph/runtime/gpu/pass/gpu_layout.cpp

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+* Copyright 2017-2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <typeindex>
+#include <typeinfo>
+
+#include "gpu_layout.hpp"
+#include "ngraph/op/replace_slice.hpp"
+#include "ngraph/runtime/gpu/gpu_op_annotations.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace gpu
+        {
+            namespace pass
+            {
+                template <>
+                void GPULayout::LAYOUT_DECL(ngraph::op::ReplaceSlice)
+                {
+                    auto rep_slice = static_cast<ngraph::op::ReplaceSlice*>(node.get());
+
+                    auto op_annotations = rep_slice->get_op_annotations();
+                    if (op_annotations)
+                    {
+                        // pass-through
+                        op_annotations->add_in_place_oi_pair({0, 0, false});
```

Comment:
`false` should be used with non-destructive ops. This is an directive for `memory_layout` that the input tensor can be reused regardless if a particular use is the last one or not. With `ReplaceSlice` which is a destructive op we probably want to only apply it to the *last* use of the input tensor.

Reply:
I see, let me change it. It will be even less ops that can be done in place.

Reply:
fixed. not bad. 
```
Benchmarking tfl/1024/mxnet-ngraph-Function_4-CompileSubgraph-fprop_cache.bprop.json, GPU backend, 10 iterations.
compile time: 13,954ms
179.3ms per iteration

---- Aggregate times per op type ----
Sum              46,527us
Dot              35,050us
Slice            18,424us
Multiply         17,302us
ReplaceSlice     13,376us
Add               8,749us
Reshape           8,516us
Subtract          5,915us
Negative          5,701us
Divide            3,510us
Broadcast         2,790us
Result              297us
OneHot               30us
```

Reply:
BTW, does it already checked here:
https://github.com/NervanaSystems/ngraph/blob/40ddf45a2ddc7f5387e1ed16cb89d1e3d506d69e/src/ngraph/pass/memory_layout.cpp#L53